### PR TITLE
[DPE-4068] Convert all tests from unittest to pytest + enable/disable secrets in every test

### PR DESCRIPTION
**Pull request by lucasgameiroborges**
Follow-up to DPE-2674 in https://github.com/canonical/postgresql-operator/pull/425

### CODE REVIEW

File: tests/unit/conftest.py [hunk 1]
```diff
@@ -30,20 +30,3 @@ def juju_has_secrets(mocker: MockerFixture):
...
-
-
-@pytest.fixture
-def only_without_juju_secrets(juju_has_secrets):
-    """Pretty way to skip Juju 2-specific tests.
-
-    Typically: to save CI time, when the same check were executed in a Juju 3-specific way already
-    """
-    if juju_has_secrets:
-        pytest.skip("Skipping legacy secrets tests")
```
Comment (lucasgameiroborges) [inline tests/unit/conftest.py:49]: We had already the whole test suite running with and without secrets (`unit-juju2` and `unit-juju3` respectively), the only change needed was to adapt the few tests that only ran in one or the other, and remove this fixtures.

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -775,298 +790,282 @@ def test_validate_config_options(self, _charm_lib):
...
-            assert (
-                "Non-existing field 'non-existing-secret' was attempted to be removed"
-                in self._caplog.text
-            )
 
-    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
-    @pytest.mark.usefixtures("use_caplog")
-    def test_delete_existing_password_secrets(self, _, __):
```
Comment (lucasgameiroborges) [inline tests/unit/test_charm.py:950]: similar to what was done in https://github.com/canonical/postgresql-operator/pull/425, this test got removed and `test_delete_password` runs with and without secrets

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -775,298 +790,282 @@ def test_validate_config_options(self, _charm_lib):
...
+@pytest.mark.parametrize("scope,is_leader", [("app", True), ("unit", True), ("unit", False)])
+@patch_network_get(private_address="1.1.1.1")
+def test_migration_from_databag(harness, juju_has_secrets, scope, is_leader):
+    """Check if we're moving on to use secrets when live upgrade from databag to Secrets usage."""
+    with (
+        patch("charm.PostgresqlOperatorCharm._on_leader_elected"),
+    ):
+        # as this test checks for a migration from databag to secrets,
+        # there's no need for this test when secrets are not enabled.
+        if not juju_has_secrets:
```
Comment (lucasgameiroborges) [inline tests/unit/test_charm.py:987]: here and on the test below, same idea from https://github.com/canonical/postgresql-operator/pull/425

File: tests/unit/conftest.py (hunk 1 above)
Comment (lucasgameiroborges) [reply tests/unit/conftest.py:49]: merging those 2 suites into one will be addressed in a follow-up PR!

File: tests/unit/test_utils.py [hunk 4]
```diff
@@ -2,19 +2,21 @@
...
 
-        second_password = new_password()
-        self.assertIsNotNone(re.fullmatch("[a-zA-Z0-9\b]{16}$", second_password))
-        self.assertNotEqual(second_password, first_password)
+def test_new_password():
+    # Test the password generation twice in order to check if we get different passwords and
+    # that they meet the required criteria.
+    first_password = new_password()
+    tc.assertEqual(len(first_password), 16)
+    tc.assertIsNotNone(re.fullmatch("[a-zA-Z0-9\b]{16}$", first_password))
```
Comment (dragomirp) [inline tests/unit/test_utils.py:18]: We can just use regular `assert`s in such cases. No need to keep the test case just for that.

Comment (lucasgameiroborges) [reply tests/unit/test_utils.py:18]: agreed, will refactor this together with merging the 2 suites in the follow-up PR.

**Pull request merged**